src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1298,86 +1298,52 @@ public static double Round(double value, int digits)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static double Round(double value, MidpointRounding mode)
         {
-            // Inline single-instruction modes
-            if (RuntimeHelpers.IsKnownConstant((int)mode))
-            {
-                if (mode == MidpointRounding.ToEven)
-                    return Round(value);
-
-                // For ARM/ARM64 we can lower it down to a single instruction FRINTA
-                // For other platforms we use a fast managed implementation
-                if (mode == MidpointRounding.AwayFromZero)
-                {
-                    if (AdvSimd.IsSupported)
-                        return AdvSimd.RoundAwayFromZeroScalar(Vector64.CreateScalar(value)).ToScalar();
+            switch (mode)
+            {
+                // Rounds to the nearest value; if the number falls midway,
+                // it is rounded to the nearest value above (for positive numbers) or below (for negative numbers)
+#pragma warning disable IntrinsicsInSystemPrivateCoreLib
```

Comment:
Let's just use the standard `if (AdvSimd.IsSupported)` check rather than `when AdvSimd.IsSupported`.

It's clearer and much more readily visible.